SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc

```diff
@@ -439,24 +440,26 @@ void HcalDigitizer::accumulateCaloHits(edm::Handle<std::vector<PCaloHit> > const
     std::vector<PCaloHit> hcalHits;
     hcalHits.reserve(hcalHitsOrig.size());
 
-    for ( unsigned int i=0; i< hcalHitsOrig.size(); i++) {
-      DetId id(hcalHitsOrig[i].id());
-      HcalDetId hid(id);
-      
-      //      if ( id.subdetId()==2 && hid.ietaAbs()<16 ) edm::LogError("HcalDigitizer") << "bad hcal id found in digitizer. Skipping " << id.rawId() << std::endl;
-      if ( !htopoP->validHcal(hid) ) edm::LogError("HcalDigitizer") << "bad hcal id found in digitizer. Skipping " << id.rawId() << std::endl;
-      else
-	hcalHits.push_back(hcalHitsOrig[i]);
-    }
     //evaluate darkening before relabeling
     if(m_HEDarkening || m_HFRecalibration){
-      darkening(hcalHits);
+      darkening(hcalHitsOrig);
```

Comment:
I don't ever plan to activate the darkening for invalid detids. It only works with HcalTestNumbering enabled. This is specified in the SLHC customizations, and it will be kept when @mark-grimes and I eventually port them to 80X.